### DISCUSSION TITLE
How to order by a projection?

### DISCUSSION THREAD

**Discussion by ethanshattuck** [Gremlinq/ExRam.Gremlinq#1276] [General]
The documentation states that you can name a projection like so:

`var dynamics = await _g
        .V<Person>()
        .Project(b => b
            .ToDynamic()
            .By(person => person.Name)
            .By(
                "count",
                __ => __
                    .Cast<object>()
                    .Out<Owns>()
                    .OfType<Pet>()
                    .Count()));`

I would like to get all people back in descending order based on their count. How can I reference the "count" projection when ordering?

`.Order(o => o.By("count")).ToArrayAsync(); //assuming something like this?`

**Comment by danielcweber** [comment]
You can select a tuple item in a `Select` operator within a `By`-modulator as [shown here](https://github.com/ExRam/ExRam.Gremlinq/blob/59f6ec5c1bc4994bf16e19a5f2518c7db768240a/test/Tests.Fixtures/QueryExecutionTest.cs#L1764). You will have to project to tuples though because you can't use dynamics in expression trees.

**Reply by ethanshattuck** [reply]
Thank you!
